### PR TITLE
CR-1097500 ASTeR TS 26 - xbutil2 - fix order of print in validate whe…

### PR DESCRIPTION
**Pull request by sadasivanshaiju**
…n P2P is enabled

### CODE REVIEW

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 1]
```diff
@@ -1025,6 +1015,9 @@ p2pTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
   auto mem_topo = reinterpret_cast<const mem_topology*>(membuf.data());
   std::string name = xrt_core::device_query<xrt_core::query::rom_vbnv>(_dev);
 
+
+  int counter = 0;
+  XBU::ProgressBar run_test("Running Test on ", mem_topo->m_count, XBU::is_escape_codes_disabled(), std::cout);
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:1020]: remove "on"

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 2]
```diff
@@ -1035,12 +1028,19 @@ p2pTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
...
     for(const auto& x : sup_list) {
       if(mem_tag.find(x) != std::string::npos && mem.m_used) {
-        if(!p2ptest_bank(_dev->get_device_handle(), _ptTest, mem_tag, static_cast<unsigned int>(midx), mem.m_base_address, mem.m_size << 10, no_dma))
-          break;
-        logger(_ptTest, "Details", mem_tag +  " validated");
+        if(!p2ptest_bank(_dev->get_device_handle(), _ptTest, mem_tag, static_cast<unsigned int>(midx), mem.m_base_address, mem.m_size << 10, no_dma)){
+           run_test.finish(false, "");
+           std::cout << EscapeCodes::cursor().prev_line() << EscapeCodes::cursor().clear_line(); 
+           break;
+	}else {
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:1035]: else on a new line

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 3]
```diff
@@ -1035,12 +1028,19 @@ p2pTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
...
+          run_test.update(++counter);
+          logger(_ptTest, "Details", mem_tag +  " validated");
+       }
       }
     }
   }
+  run_test.finish(true, "");
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:1042]: this needs to be wrapped in an if statement. Please remove `run_test.finish(false, "");` L1032-1033 and do the following:
```suggestion
  if(counter < mem_topo->m_count)
    run_test.finish(false, "");
  else
    run_test.finish(true, "");
```

**Review by AShivangi — CHANGES_REQUESTED** [review]
Please make the requested changes and add sample outputs for:
1. passed test case
2. failed test case
before re-requesting review.

**Comment by sadasivanshaiju** [comment]
Running Test on : ..........25%.........50%.........75%....
Test 1 [0000:b1:00.1]     : Peer to peer bar
    Details               : HBM[0] validated
                            HBM[1] validated
                            HBM[2] validated
                            HBM[3] validated
                            HBM[4] validated
                            HBM[5] validated
                            HBM[6] validated
                            HBM[7] validated
                            HBM[8] validated
                            HBM[9] validated
                            HBM[10] validated
                            HBM[11] validated
                            HBM[12] validated
                            HBM[13] validated
                            HBM[14] validated
                            HBM[15] validated
                            HBM[16] validated
                            HBM[17] validated
                            HBM[18] validated
                            HBM[19] validated
                            HBM[20] validated
                            HBM[21] validated
                            HBM[22] validated
                            HBM[23] validated
                            HBM[24] validated
                            HBM[25] validated
                            HBM[26] validated
                            HBM[27] validated
                            HBM[28] validated
                            HBM[29] validated
                            HBM[30] validated
                            HBM[31] validated
                             validated
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Validation completed. Please run the command '--verbose' option

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 4]
```diff
@@ -1035,12 +1028,24 @@ p2pTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
     const std::string mem_tag(reinterpret_cast<const char *>(mem.m_tag));
     for(const auto& x : sup_list) {
       if(mem_tag.find(x) != std::string::npos && mem.m_used) {
-        if(!p2ptest_bank(_dev->get_device_handle(), _ptTest, mem_tag, static_cast<unsigned int>(midx), mem.m_base_address, mem.m_size << 10, no_dma))
-          break;
-        logger(_ptTest, "Details", mem_tag +  " validated");
+        if(!p2ptest_bank(_dev->get_device_handle(), _ptTest, mem_tag, static_cast<unsigned int>(midx), mem.m_base_address, mem.m_size << 10, no_dma)){
+           run_test.finish(false, "");
+           std::cout << EscapeCodes::cursor().prev_line() << EscapeCodes::cursor().clear_line();
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:1033]: please remove these lines

**Review by AShivangi — APPROVED** [review]

**Pull request merged**